### ISSUE TITLE
Enhancement: Enable use of Readable stream when uploading attachments

### ISSUE THREAD

**Issue by csharpsi** [XeroAPI/xero-node#631]
**SDK you're using (please complete the following information):**
 - Version 4.17.0

**Is your feature request related to a problem? Please describe.**
Currently, the accounting API allows for attaching documents by using a file stream. This relies on the file existing on disk in order to use the `fs.ReadStream` API. 

Because of this restriction, when auto generating a document receipt in memory, it is necessary to write the file to disk before we can use the `fs.createReadStream(...)` method to pass the file onto the Xero Accounting API client (in my specific case the `updateInvoiceAttachmentByFileName` method).

**Describe the solution you'd like**
The `readStream` argument uses the type `fs.ReadStream`. I propose that we define this argument as a union type of `fs.ReadStream | Readable | Buffer` or define an overloaded method (to maintain backward compatibility) and use the [Stream](https://nodejs.org/docs/latest-v14.x/api/stream.html) API to enable the use of `Readable` streams. An example of this would look like:

```typescript
import { Readable } from "stream";
...

updateInvoiceAttachmentByFileName(xeroTenantId: string, invoiceID: string, fileName: string, body: fs.ReadStream | Readable | Buffer, options?: {
    headers: {
        [name: string]: string;
    };
}): Promise<{
    response: http.IncomingMessage;
    body: Attachments;
}>;
```

**Describe alternatives you've considered**
The only alternative currently is to temporarily write the buffer to disk before reading it into an FS stream, which is pretty inefficient.

**Additional Context**
I'd be happy to submit a PR for this if someone could point me in the right direction. It doesn't look super clear how this would be done using the Open API generator

**Comment by mikemklee** [comment]
Any updates to this? Is the approach described above to temporarily write the buffer to disk before reading it again as a FS stream still the only workaround?

**Comment by sangeet-joy-tw** [comment]
The changes to accept Buffer & Readable stream type has been released with the new version **v5.1.0**

Please let us know with any more issues.

@mikemklee @csharpsi

**Comment by sangeet-joy-tw** [comment]
The changes to accept Buffer & Readable stream type has been released with the new version **v5.1.0**

Please let us know with any more issues.

@mikemklee @csharpsi

**Comment by csharpsi** [comment]
Hi @sangeet-joy-tw 

Thanks for the update! However, I don't think it's working. 

When I perform an export, the file is not attached to the invoice. I have a unit test that proves that the document stream passed into the `updateInvoiceAttachmentByFileName` function matches the byte array of the test file and that is passing, so I am confident that the issue is with the SDK.  

Example effected invoices:
* `22e3d6d4-3442-444b-b109-fa359bcacef6`
* `3c425103-1d67-4189-9b1f-3ef1baf286e1`

The below examples were imported into Xero, the first using the latest version of the xero-node SDK and the second using version 4.17.0 

![Latest Version](https://github.com/XeroAPI/xero-node/assets/1210114/73863216-ae87-4685-a232-7ef4fb27e249)

![Version 4.17.0](https://github.com/XeroAPI/xero-node/assets/1210114/64c1af5a-4bf4-47ac-8f78-83bfb2e6c08c)

Additionally, I am seeing a console.log message `JSON parse body failed` which looks to be coming from the SDK [here](https://github.com/XeroAPI/xero-node/blob/master/src/gen/api/accountingApi.ts#L17719)

![image](https://github.com/XeroAPI/xero-node/assets/1210114/cd4f924c-fc53-4c94-bd96-1f8c0f64c3bb)

If I can help any further please let me know
